sdk/src/main/java/com/hedera/hashgraph/sdk/Network.java

```diff
@@ -234,6 +234,12 @@ synchronized Map<String, AccountId> getNetwork() {
 
     @Override
     protected Node createNodeFromNetworkEntry(Map.Entry<String, AccountId> entry) {
+        if (addressBook != null && !addressBook.isEmpty()) {
+            return new Node(entry.getValue(), entry.getKey(), executor)
+                    .setVerifyCertificates(verifyCertificates)
+                    .setAddressBookEntry(addressBook.get(entry.getValue()));
+        }
+
         return new Node(entry.getValue(), entry.getKey(), executor).setVerifyCertificates(verifyCertificates);
```

Comment:
This doesn't handle another corner case where the address book has `N` entries and gets updated to have `N-X` entries e.g. it shrinks in size. If the previously set address book entry goes missing or the entire address book is cleared then it should also clear the corresponding `node.addressBookEntry` values.

```suggestion
        var addressBookEntry = addressBook != null ? addressBook.get(entry.getValue()) : null;
        return new Node(entry.getValue(), entry.getKey(), executor)
            .setAddressBookEntry(addressBookEntry)
            .setVerifyCertificates(verifyCertificates);
```

Reply:
Note taken. I've made the necessary changes. Thanks for your input.